modules/network/src/integrationTest/java/org/apache/ignite/network/scalecube/ItScaleCubeNetworkMessagingTest.java

```diff
@@ -355,25 +357,44 @@ public void nodeCannotReuseOldId(TestInfo testInfo) throws Exception {
     }
 
     private void knockOutNode(String outcastName) throws InterruptedException {
-        CountDownLatch disappeared = new CountDownLatch(1);
+        CountDownLatch disappeared = new CountDownLatch(2);
 
-        testCluster.members.get(0).topologyService().addEventHandler(new TopologyEventHandler() {
+        TopologyEventHandler disappearListener = new TopologyEventHandler() {
             @Override
             public void onDisappeared(ClusterNode member) {
                 if (Objects.equals(member.name(), outcastName)) {
                     disappeared.countDown();
                 }
             }
-        });
+        };
 
-        testCluster.members.stream()
+        List<ClusterService> notOutcasts = testCluster.members.stream()
```

Comment:
I prefer setting event handler prior to dropping messages, because dropping messages from node1 to node2 may trigger "Node left event" on node3 because of the scalecube's nature